src/libs/Trie/index.js

```diff
@@ -0,0 +1,121 @@
+import _ from 'underscore';
+import TrieNode from './TrieNode';
+
+class Trie {
+    constructor() {
+        this.root = new TrieNode();
+    }
+
+    /**
+    * Add a word to the Trie
+    * @param {String} word
+    * @param {Object} [metaData] - attach additional data to the word
+    * @param {TrieNode} newNode
+    * @param {Boolean} [containChar] - empty word doesn't have any char
+    * @returns {void}
+    */
+    add(word, metaData, newNode = this.root, containChar = false) {
```

Comment:
There's no valid case where an external method should pass `containChar = true`, because we don't want any empty words inserted in our Trie. So what we should do is create a new private function called `addHelper`, `addInternal`, `_add`, or something similar. There are a few options to make private methods:

1. Use [private member syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields) ❌ We can't do this because it's a language feature our team has not yet agreed to use (we can't use TypeScript either 😭)
1. Use an IIFE (immediately invoked function expression) to make it so that the private method is not available on the class prototype, and therefore can't be used by anyone who imports the class.

    ```js
    const Trie = (function() {
        /**
         * Private helper function to add a word to the Trie.
         * @param {String} word
         * @param {Object} [metaData]
         * @param {TrieNode} [newNode]
         */
        function addHelper(word, metaData, newNode) {
            // In here, we know that an empty word means that we've recursively reached
            // the end of a real word, and we should `setCompleteWord`
        }
        
        return class {
            // Here's the rest of the class, including the constructor and public functions
            ...
            ...
            
            /**
             * Add a word to the Trie
             * @param {String} word
             * @param {Object} [metaData] - attach additional data to the word
             */
            add(word, metaData) {
                if (word.length === 0) {
                    throw new Error('Cannot insert empty word into Trie');
                }
                
                addHelper(word, metaData, this.root);
            }
        }
    })();
    
    export default Trie;
    ```
        
So I think we should use the IIFE approach. This way, there's a simpler external interface that can't be easily misused.

Reply:
I guess another alternative would be to just rename `containChar` to `allowEmptyWords` and assume nobody will pass `true` for that param without knowing what they're doing. That's probably a simpler way forward than the fancy solution with an IIFE

Reply:
Done.